### PR TITLE
Fix import errors for configs and plugins omitting `/index.js`

### LINKED ISSUE #7576: Compatibility issue between `stylelint@16.3.0` and `stylelint-prettier@5.0.0`
### What minimal example or steps are needed to reproduce the bug?

 Having stylelint@16.3.0 and stylelint-prettier@5.0.0 in the same repo.

### What minimal configuration is needed to reproduce the bug?

 Seems to be a compatibility issue between `stylelint@16.3.0` and `stylelint-prettier@5.0.0`.
Capture d'écran 2024-03-25 104853

### How did you run Stylelint?

`yarn stylelint`

### Which Stylelint-related dependencies are you using?

```json
    "stylelint": "16.3.0",
    "stylelint-config-standard": "36.0.0",
    "stylelint-prettier": "5.0.0",
```

### What did you expect to happen?

The linting to work.

### What actually happened?

Capture d'écran 2024-03-25 104853

### Do you have a proposal to fix the bug?

_No response_

### LINKED ISSUE #7577: Error upgrading from v16.2.1 to v16.3.0: No configuration provided
### What minimal example or steps are needed to reproduce the bug?

upon upgrading from `v16.2.1` -> `v16.3.0` (no other code/dependencies were changed), i get this error when linting files with stylelint

```
yarn stylelint --fix --allow-empty-input '**/*.{vue,css,scss}'

Error: No configuration provided for /app/src/app.vue
    at configurationError (file:///app/node_modules/stylelint/lib/utils/configurationError.mjs:12:49)
    at getConfigForFile (file:///app/node_modules/stylelint/lib/getConfigForFile.mjs:67:4)
    at async isPathIgnored (file:///app/node_modules/stylelint/lib/isPathIgnored.mjs:23:17)
    at async lintSource (file:///app/node_modules/stylelint/lib/lintSource.mjs:43:20)
    at async file:///app/node_modules/stylelint/lib/standalone.mjs:214:27
    at async Promise.all (index 0)
    at async standalone (file:///app/node_modules/stylelint/lib/standalone.mjs:257:22)
error Command failed with exit code 78.

```

### What minimal configuration is needed to reproduce the bug?

```
// stylelint.config.js
module.exports = {
  extends: [
    'stylelint-config-standard-scss',
    'stylelint-config-standard-vue/scss',
    'stylelint-config-recess-order',
    'stylelint-config-css-modules',
  ],
  rules: {},
}
```

### How did you run Stylelint?

`yarn stylelint --fix --allow-empty-input '**/*.{vue,css,scss}'`

### Which Stylelint-related dependencies are you using?

```
    "stylelint": "16.3.0",
    "stylelint-config-css-modules": "^4.1.0",
    "stylelint-config-recess-order": "^4.0.0",
    "stylelint-config-standard-scss": "^12.0.0",
    "stylelint-config-standard-vue": "^1.0.0",
    "stylelint-scss": "^6.0.0",
```

### What did you expect to happen?

No error upon the minor upgrade from `v16.2.1` -> `v16.3.0` (no other dependencies were changed)

### What actually happened?

```
yarn stylelint --fix --allow-empty-input '**/*.{vue,css,scss}'

Error: No configuration provided for /app/src/app.vue
    at configurationError (file:///app/node_modules/stylelint/lib/utils/configurationError.mjs:12:49)
    at getConfigForFile (file:///app/node_modules/stylelint/lib/getConfigForFile.mjs:67:4)
    at async isPathIgnored (file:///app/node_modules/stylelint/lib/isPathIgnored.mjs:23:17)
    at async lintSource (file:///app/node_modules/stylelint/lib/lintSource.mjs:43:20)
    at async file:///app/node_modules/stylelint/lib/standalone.mjs:214:27
    at async Promise.all (index 0)
    at async standalone (file:///app/node_modules/stylelint/lib/standalone.mjs:257:22)
error Command failed with exit code 78.

```

### Do you have a proposal to fix the bug?

_No response_

### PR DESCRIPTION
**Pull request by ota-meshi**
> Which issue, if any, is this issue related to?

Closes #7576
Closes #7577

Ref #7532

> Is there anything in the PR that needs further explanation?

This PR fixes an issue where cannot resolve paths in configurations that expect `index.js` to resolve implicitly by Node.js.

### CODE REVIEW

File: .changeset/three-pumas-help.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fix an issue where config for `/index.js` could not be loaded
```
Comment (Mouvedia) [inline .changeset/three-pumas-help.md:5]: Something like

```suggestion
Fixed: ensure that `resolveSilent` util resolves `.` as `index.js`
```

would be more accurate.

**Review by Mouvedia — COMMENTED** [review]
Thanks for the quick fix.

File: lib/utils/resolveSilent.mjs [hunk 2]
```diff
@@ -15,6 +15,15 @@ const pathSuffixes = ['', '.js', '.json', `${path.sep}index.js`, `${path.sep}ind
 
 const specifierSuffixes = ['', '.js', '.json', '/index.js', '/index.json'];
 
+/**
+ * Checks whether the given file exists as a file.
+ * @param {string} filename The file to check.
+ * @returns {boolean} `true` if the file exists and is a file, otherwise `false`.
+ */
+function existsFile(filename) {
+	return fs.existsSync(filename) && fs.statSync(filename).isFile();
```
Comment (ybiquitous) [inline lib/utils/resolveSilent.mjs:24]: [suggest] Minor. How about using the `throwIfNoEntry` option like this? I can reduce one `fs` call.

```suggestion
	return fs.statSync(filename, { throwIfNoEntry: false })?.isFile() ?? false;
```

File: .changeset/three-pumas-help.md (hunk 1 above)
Comment (ybiquitous) [inline .changeset/three-pumas-help.md:5]: [suggest] Let's adjust the changelog for consistency:

```suggestion
Fixed: import errors for configs and plugins omitting `/index.js`
```

**Review by ybiquitous — COMMENTED** [review]
@ota-meshi Thank you for the quick fix! Overall LGTM, but I've left a few comments. Can you take a look?

File: .changeset/three-pumas-help.md (hunk 1 above)
Comment (ybiquitous) [reply .changeset/three-pumas-help.md:5]: Thanks for the suggestion. But I think `resolveSilent` is internal, so I'd like to avoid it if possible.

**Comment by ota-meshi** [comment]
Thank you for the suggestions! I've modified this PR.

**Review by ybiquitous — APPROVED** [review]
Thank you. LGTM 👍🏼

**Pull request merged**